C++/include/algorithm/dynamic_programming/longest_decreasing_subsequence.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef LONGEST_DECREASING_SUBSEQUENCE
+#define LONGEST_DECREASING_SUBSEQUENCE
+
+/*
+Longest Decreasing Subsequence
+---------------------
+This algorithm finds the longest decreasing  subsequence in a sequence of numbers
+It follows the principals of dynamic programming
+
+Time complexity
+----------------
+O(n logn) where n is the number of elements in the initial list
+
+Space complexity
+---------------
+O(n)
+
+*/
+
+#include <iostream>
+#include <vector>
+
+int longest_decreasing_subsequence(std::vector<int>& input,  const bool to_show_state = true) {//order of 1 means increasing subsequence, -1 for decreasing subsequence
```

Comment:
Please explain what the functions do. You can use [this](https://github.com/faheel/Algos/blob/reorganise/C%2B%2B/include/algorithm/backtracking/n_queens.hpp) as a guide.

Reply:
Also, your comment below no longer reflects the function signature.